### PR TITLE
[BUGFIX] Use string dialect name if not found in enum

### PR DESCRIPTION
**Pull request by anthonyburdi**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- Dialects not found in GESqlDialect were causing errors e.g. Vertica. This PR is intended as a quick fix to ensure the string name is used if the dialect is not found in the enum.
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [ ] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [ ] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

**Review by Kilo59 — APPROVED** [review]

**Review by alexsherstinsky — APPROVED** [review]
LGTM

**Comment by anthonyburdi** [comment]
Holding to implement feedback from @billdirks @Kilo59 @alexsherstinsky

File: great_expectations/execution_engine/sqlalchemy_batch_data.py [hunk 1]
```diff
@@ -109,7 +109,13 @@ def __init__(
                 "schema_name can only be used with table_name. Use temp_table_schema_name to provide a target schema for creating a temporary table."
             )
 
-        dialect: GESqlDialect = GESqlDialect(engine.dialect.name.lower())
+        dialect_name: str = engine.dialect.name.lower()
+
+        try:
+            dialect: GESqlDialect = GESqlDialect(dialect_name)
+        except ValueError:
+            dialect: GESqlDialect = GESqlDialect.UNSUPPORTED
```
Comment (billdirks) [inline great_expectations/execution_engine/sqlalchemy_batch_data.py:116]: If you want to store the unsupported dialect name you could hang it off dialect as a quick hack. That is follow this with the line `dialect._unsupported_dialect = dialect_name`. You would want to do that below also. If we end up doing anything with this value, we should do something better as a followup.

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM to me as well :)

File: great_expectations/execution_engine/sqlalchemy_batch_data.py (hunk 1 above)
Comment (Kilo59) [reply great_expectations/execution_engine/sqlalchemy_batch_data.py:116]: Another idea (not needed now)
Would be do do this logic within `_missing_` which gets called when the value isn't a member.
Which also means we could get rid of this try: expect.

https://docs.python.org/3/library/enum.html#supported-sunder-names

**Review by billdirks — APPROVED** [review]
Thanks for making this change! LGTM.

**Pull request merged**